### PR TITLE
Replace PHP explanation with cURL by explanations with the SRB-PHP lib

### LINKED ISSUE #3: Migrate the API documentation from naked PHP to PHP with our library

### PR DESCRIPTION
**Pull request by PaulGirardin**
Fixes #3 

!!!!! STOP AND READ !!!!!

If the dropdown above says "base fork: lord/master", you are submitting your change to ALL USERS OF SLATE, not just your company. This is probably not what you want. Click "base fork" to change it to the right place.

If you're actually trying to submit a change to upstream Slate, please submit to our dev branch, PRs sent to the master branch are generally rejected.

### CODE REVIEW

File: source/includes/api/php.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Using PHP
+
+If you are coding in PHP, you can use our [PHP library](https://shoprunback.github.io/documentation/php.html). You don't have to code your own API wrapper!
```
[the remark this answers is not in the document]
Comment (PaulGirardin) [reply source/includes/api/php.md]: J'ai fait une section à part pour que les lecteurs le voient directement et qu'ils ne cherchent pas l'info

File: source/includes/api/catalog/_brand.md [hunk 2]
```diff
@@ -124,43 +94,24 @@ curl -X "GET" "https://dashboard.shoprunback.com/api/v1/brands" \
...
-// Set method
-curl_setopt($ch, CURLOPT_CUSTOMREQUEST, 'GET');
+// Set your token
+\Shoprunback\RestClient::getClient()->setToken('your_token');
 
-// Set options
-curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
+// Get all your brands
+$brands = \Shoprunback\Elements\Brand::all();
+$brands[0]; // Returns the Brand with the index 0
```
Comment (leo-martin) [inline source/includes/api/catalog/_brand.md]: we dont need this, the endpoint is to List all the brands, not to get index[0] or index[1]
you can remove the code after `$brands = \Shoprunback\Elements\Brand::all();`

File: source/includes/api/catalog/_brand.md [hunk 3]
```diff
@@ -124,43 +94,24 @@ curl -X "GET" "https://dashboard.shoprunback.com/api/v1/brands" \
...
-
-if(!$resp) {
-  die('Error: "' . curl_error($ch) . '" - Code: ' . curl_errno($ch));
-} else {
-  echo "Response HTTP Status Code : " . curl_getinfo($ch, CURLINFO_HTTP_CODE);
-  echo "\nResponse HTTP Body : " . $resp;
+try {
+  $brands[1]; // Returns the Brand with the index 1 (if it exists)
+} catch (\Shoprunback\Error\ElementIndexDoesntExists $e) {
+  // If it doesn't exist, it returns an Exception
```
Comment (leo-martin) [inline source/includes/api/catalog/_brand.md]: same comment

File: source/includes/api/catalog/_product.md [hunk 4]
```diff
@@ -391,42 +326,24 @@ curl -X "GET" "https://dashboard.shoprunback.com/api/v1/products?page=1" \
...
-if(!$resp) {
-  die('Error: "' . curl_error($ch) . '" - Code: ' . curl_errno($ch));
-} else {
-  echo "Response HTTP Status Code : " . curl_getinfo($ch, CURLINFO_HTTP_CODE);
-  echo "\nResponse HTTP Body : " . $resp;
+try {
+  $products[0]; // Returns the Product with the index 0 (if it exists)
+  $products[1]; // Returns the Product with the index 1 (if it exists)
+} catch (\Shoprunback\Error\ElementIndexDoesntExists $e) {
+  // If it doesn't exist, it returns an Exception
```
Comment (leo-martin) [inline source/includes/api/catalog/_product.md]: same comment than brands, we are fetching the list of products.

File: source/includes/api/order.md [hunk 5]
```diff
@@ -305,7 +275,26 @@ curl -X "GET" "https://dashboard.shoprunback.com/api/v1/orders?page=1" \
...
+
+// Get all your orders
+$orders = \Shoprunback\Elements\Order::all();
+
+try {
+  $orders[0]; // Returns the Order with the index 0 (if it exists)
+  $orders[1]; // Returns the Order with the index 1 (if it exists)
+} catch (\Shoprunback\Error\ElementIndexDoesntExists $e) {
+  // If it doesn't exist, it returns an Exception
+}
```
Comment (leo-martin) [inline source/includes/api/order.md]: same comment as brand and product

File: source/includes/api/order.md [hunk 6]
```diff
@@ -362,195 +351,3 @@ curl -X "GET" "https://dashboard.shoprunback.com/api/v1/orders?page=1" \
...
-
-
-
-
-
-
-
-
-
-## Update an order
```
Comment (leo-martin) [inline source/includes/api/order.md]: where did this go?

Comment (PaulGirardin) [reply source/includes/api/order.md]: The PUT endpoint for orders has been removed

File: source/includes/api/catalog/_brand.md [hunk 7]
```diff
@@ -1,8 +1,5 @@
 ## Create a brand
 
-By default, once your retailer account is created and your company details entered, a default brand is created.
-But you can add your own brands if you have multiple brands in your catalog.
-
```
[the remark this answers is not in the document]
Comment (PaulGirardin) [reply source/includes/api/catalog/_brand.md]: It's still here, it has been moved to line 58 to be aligned with the right-column

Before:
doc-api-create-brand-not-aligned


After:
doc-api-create-brand-aligned

File: source/includes/api/catalog/_product.md [hunk 8]
```diff
@@ -232,61 +191,31 @@ curl -X "POST" "https://dashboard.shoprunback.com/api/v1/products" \
...
-}
-
-// Close request to clear up some resources
-curl_close($ch);
-?>
+// Load the library
+require 'path/to/lib/shoprunback-php/init.php';
+
+// Set your token
+\Shoprunback\RestClient::getClient()->setToken('your_token');
```
Comment (tildedash) [inline source/includes/api/catalog/_product.md]: isn't it needed to set the API url / environment ?

File: source/includes/api/catalog/_product.md [hunk 9]
```diff
@@ -232,61 +191,31 @@ curl -X "POST" "https://dashboard.shoprunback.com/api/v1/products" \
...
+$product->brand = $brand;
+$product->ean = '1258987561456';
+$product->picture_url = 'https://s3.eu-central-1.amazonaws.com/shoprunback-dev/products/484/7b4/f4-/pictures/medium.?1505912197';
+
+// Save the Product
+$product->save();
 ```
 
-> The above command returns the same JSON object with the id of the created product:
+> The above command (except for PHP) returns the same JSON object with the id of the created product:
```
Comment (tildedash) [inline source/includes/api/catalog/_product.md]: not needed to talk about this particularity of PHP

File: source/includes/api/catalog/_product.md [hunk 10]
```diff
@@ -304,6 +233,12 @@ curl_close($ch);
 }
 ```
 
+You can create your product directly with the brand object embedded.
+
+If 2 products have the same brand's `reference`, only one brand object is created.
```
Comment (tildedash) [inline source/includes/api/catalog/_product.md]: you can remove it

File: source/includes/api/catalog/_product.md [hunk 11]
```diff
@@ -304,6 +233,12 @@ curl_close($ch);
...
+
+The image in not mandatory but the return experience of your customer will be better with it.
```
Comment (tildedash) [inline source/includes/api/catalog/_product.md]: the return experience is better with it

File: source/includes/api/catalog/_product.md [hunk 12]
```diff
@@ -391,42 +326,17 @@ curl -X "GET" "https://dashboard.shoprunback.com/api/v1/products?page=1" \
...
 
-// Close request to clear up some resources
-curl_close($ch);
-?>
+// Get all your products
+$products = \Shoprunback\Elements\Product::all();
 ```
 
-> The above command returns JSON structured like this:
+> The above command (except for PHP) returns JSON structured like this:
```
Comment (tildedash) [inline source/includes/api/catalog/_product.md]: don't talk about PHP, it's a library not a restclient, of course there is a object as response and not a JSON.

File: source/includes/api/introduction.md [hunk 13]
```diff
@@ -12,4 +12,6 @@ You can also get the swagger documentation on [https://app.swaggerhub.com/apis/S
 
 For feature request, use the built in form: [https://dashboard.shoprunback.com/features](https://dashboard.shoprunback.com/features)
 
-If you got any questions, send an email to: julien_at_shoprunback.com.
+If you are coding in PHP, you can use our [PHP library](https://shoprunback.github.io/documentation/php.html).
```
Comment (tildedash) [inline source/includes/api/introduction.md]: Use our PHP Library

File: source/includes/api/order.md [hunk 14]
```diff
@@ -94,76 +89,46 @@ curl -X "POST" "https://dashboard.shoprunback.com/api/v1/orders" \
...
+$order->customer = $customer;
+$order->items = $items;
+$order->ordered_at = '2017-02-03';
+
+// We save the Order
+$order->save();
 ```
 
-> The above command returns the same JSON object with the id of the created order, customer and items:
+> The above command (except for PHP) returns the same JSON object with the id of the created order, customer and items:
```
Comment (tildedash) [inline source/includes/api/order.md]: - PHP

File: source/includes/api/catalog/_product.md (hunk 8 above)
Comment (PaulGirardin) [reply source/includes/api/catalog/_product.md]: It is indeed, I added it just below

File: source/includes/api/catalog.md [hunk 15]
```diff
@@ -1,6 +1,6 @@
 # Products catalog
 
 Once your account created and configured (follow the onboarding process for that), you must push your products catalog.
-Only product in the ShopRunBack catalog can be returned.
+Only products in the ShopRunBack catalog can be returned.
 
-First, create your brands if you have any and then create your products.
\ No newline at end of file
+First, create your brand(s) if you don't have any and then create your products.
```
Comment (tildedash) [inline source/includes/api/catalog.md]: no, it is not required to create brand. With your update, the user will think that he must create brands.

File: source/includes/api/catalog/_product.md [hunk 16]
```diff
@@ -132,7 +100,7 @@ picture_file_url | no | public URL to the product image (JPG or PNG), to avoid i
 picture_file_base64 | no | if your product's image is not hosted on the internet, you can provide it with a base64 version of it
 
 <aside class="success">
-If you don't have more than one brand, you don't have to provide the brand_id.
+If you only have one brand, you don't have to provide the brand_id.
```
Comment (tildedash) [inline source/includes/api/catalog/_product.md]: If you only have the default brand, …

File: source/includes/api/catalog/_brand.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (tildedash) [reply source/includes/api/catalog/_brand.md]: OK

**Pull request merged**